sdk/identity/azure-identity-v2/src/main/java/com/azure/identity/v2/AzureToolkitCredential.java

```diff
@@ -0,0 +1,107 @@
+import com.azure.identity.v2.implementation.models.MsalToken;
+import com.azure.identity.v2.implementation.models.PublicClientOptions;
+import com.azure.identity.v2.implementation.util.IdentityConstants;
+import com.azure.identity.v2.implementation.util.LoggingUtil;
+import com.azure.v2.core.credentials.TokenCredential;
+import com.azure.v2.core.credentials.TokenRequestContext;
+import io.clientcore.core.credentials.oauth.AccessToken;
+import io.clientcore.core.instrumentation.logging.ClientLogger;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * <p>IntelliJ IDEA is an integrated development environment (IDE) developed by JetBrains, which provides a variety of
+ * features to support software development, such as code completion, debugging, and testing.
+ * Azure offers <a href="https://learn.microsoft.com/azure/developer/java/toolkit-for-intellij/">Azure Toolkit
+ * for IntelliJ plugin</a> for the IntelliJ IDEA development environment. It
+ * enables developers to create, test, and deploy Java applications to the Azure cloud platform. In order to
+ * use the plugin authentication as a user or service principal against
+ * <a href="https://learn.microsoft.com/entra/fundamentals/">Microsoft Entra ID</a> is required.
+ * The IntelliJCredential authenticates in a development environment and acquires a token on behalf of the
+ * logged-in account in Azure Toolkit for IntelliJ. It uses the logged in user information on the IntelliJ IDE and uses
+ * it to authenticate the application against Microsoft Entra ID.</p>
+ *
+ * <h2>Configure IntelliJCredential</h2>
+ *
+ * <p>Follow the steps outlined below:</p>
+ *
+ * <ol>
+ *     <li>In your IntelliJ window, open File > Settings > Plugins.</li>
+ *     <li>Search for "Azure Toolkit for IntelliJ" in the marketplace. Install and restart IDE.</li>
+ *     <li>Find the new menu item Tools > Azure > Azure Sign In.</li>
+ *     <li>Device Login will help you log in as a user account. Follow the instructions to log in on the
+ *     login.microsoftonline.com website with the device code. IntelliJ will prompt you to select your subscriptions.
+ *     Select the subscription with the resources that you want to access.</li>
+ * </ol>
+ *
+ * <p> Once the developer has followed the steps above and authenticated successfully with
+ * Azure Tools for IntelliJ plugin in the IntelliJ IDE then this credential can be used in the development code to
+ * reuse the cached plugin credentials.</p>
+ *
+ * <p><strong>Sample: Construct IntelliJCredential</strong></p>
+ *
+ * <p>The following code sample demonstrates the creation of a {@link AzureToolkitCredential},
+ * using the {@link AzureToolkitCredentialBuilder} to configure it. Once this credential is
+ * created, it may be passed into the builder of many of the Azure SDK for Java client builders as the 'credential'
+ * parameter.</p>
+ *
+ * <pre>
+ * TokenCredential intelliJCredential = new IntelliJCredentialBuilder&#40;&#41;.build&#40;&#41;;
```

Comment:
Need to pull in samples project (I went looking for that to add this fix to):

```suggestion
 * TokenCredential azureToolkitCredential = new AzureToolkitCredentialBuilder&#40;&#41;.build&#40;&#41;;
```